components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -0,0 +1,438 @@
+import org.wso2.charon3.core.exceptions.NotImplementedException;
+import org.wso2.charon3.core.extensions.RoleManager;
+import org.wso2.charon3.core.objects.Group;
+import org.wso2.charon3.core.objects.Role;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.utils.codeutils.ExpressionNode;
+import org.wso2.charon3.core.utils.codeutils.Node;
+import org.wso2.charon3.core.utils.codeutils.OperationNode;
+import org.wso2.charon3.core.utils.codeutils.SearchRequest;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+
+/**
+ * Implementation of the {@link RoleManager} interface.
+ */
+public class SCIMRoleManager implements RoleManager {
+
+    private static final Log log = LogFactory.getLog(SCIMRoleManager.class);
+    private RoleManagementService roleManagementService;
+    private String tenantDomain;
+    private static final String FILTERING_DELIMITER = "*";
+
+    public SCIMRoleManager(RoleManagementService roleManagementService, String tenantDomain) {
+
+        this.roleManagementService = roleManagementService;
+        this.tenantDomain = tenantDomain;
+    }
+
+    @Override
+    public Role createRole(Role role) throws CharonException, ConflictException, BadRequestException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Creating role: " + role.getDisplayName());
+        }
+        try {
+            // Check if the role already exists.
+            if (roleManagementService.isExistingRole(role.getId(), tenantDomain)) {
+                String error = "Role with name: " + role.getDisplayName() + " already exists in the system.";
+                throw new ConflictException(error);
+            }
+            RoleBasicInfo roleBasicInfo = roleManagementService
+                    .addRole(role.getDisplayName(), role.getUsers(), role.getGroups(), role.getPermissions(), tenantDomain);
+
+            Role createdRole = new Role();
+            createdRole.setId(roleBasicInfo.getId());
+            String locationURI = SCIMCommonUtils.getSCIMRoleURL(roleBasicInfo.getId());
+            createdRole.setLocation(locationURI);
+            createdRole.setDisplayName(roleBasicInfo.getName());
+            createdRole.setSchemas();
+
+            return createdRole;
+        } catch (IdentityRoleManagementException e) {
+            throw new CharonException(
+                    String.format("Error occurred while adding a new role: %s", role.getDisplayName()), e);
+        }
+    }
+
+    @Override
+    public Role getRole(String roleID, Map<String, Boolean> requiredAttributes)
+            throws BadRequestException, CharonException {
+
+        try {
+            org.wso2.carbon.identity.role.mgt.core.Role role = roleManagementService.getRole(roleID, tenantDomain);
```

Comment:
let's return not found exception if role is null

Reply:
Fixed and updated the PR.